clearwater-config-manager.root/usr/share/clearwater/clearwater-config-manager/scripts/print-bgcf-configuration

```diff
@@ -0,0 +1,96 @@
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import json
+
+# This does some basic validation of the BGCF configuration file, and
+# prints the contents
+try:
+    with open("/etc/clearwater/bgcf.json") as bgcf_file:
+        try:
+            bgcf_data = json.load(bgcf_file)
+
+            # The expected format is:
+            #
+            # {
+            #   "routes" :
+            #   [
+            #     {
+            #       "name" : "<route 1 descriptive name>",
+            #       "<domain|number>" : "<SIP trunk IP address, domain name or routing number>",
+            #       "route" : [<"IBCF SIP URI">, <"IBCF SIP URI">,...]
+            #     },
+            #     ...
+            #   ]
+            # }
+            routes = bgcf_data["routes"]
+
+            try:
+                for bgcf_route in routes:
+                    name = bgcf_route["name"]
+                    route = bgcf_route["route"]
+                    value = ""
+
+                    try:
+                        domain = bgcf_route["domain"]
+                        value = "Domain: " + domain
+                    except KeyError as e:
+                        pass
+
+                    try:
+                        number = bgcf_route["number"]
+                        value = "Number: " + number
+                    except KeyError as e:
+                        pass
```

Comment:
Assuming that we want to enforce the json contains only one of domain|number per route, we should check that we don't have both here. As i read it, if both were configured, we would just overwrite the Domain entry with the Number entry.